pulp_smart_proxy/app/views.py

```diff
@@ -0,0 +1,69 @@
+from urllib.parse import urljoin
+
+from django.conf import settings
+from drf_spectacular.utils import extend_schema
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from pulpcore.app.apps import pulp_plugin_configs, get_plugin_config
+
+
+class FeaturesView(APIView):
+    """
+    Returns features of the smart_proxy
+    """
+
+    @extend_schema(
+        summary="Inspect features",
+        operation_id="features_read",
+    )
+    def get(self, request):
+        data = ["pulpcore"]
+        return Response(data)
+
+
+class FeaturesV2View(APIView):
+    """
+    Returns features of the smart_proxy in v2 format
+    """
+
+    @extend_schema(
+        summary="Inspect features",
+        operation_id="featuresv2_read",
+    )
+    def get(self, request):
+        host = settings.CONTENT_ORIGIN or request.get_host()
+        capabilities = [app.label for app in pulp_plugin_configs()]
+        data = {
+            "pulpcore": {
+                "http_enabled": False,
+                "https_enabled": True,
+                "settings": {
+                    "pulp_url": host,
+                    "mirror": False,
+                    "content_app_url": urljoin(host, settings.CONTENT_PATH_PREFIX),
+                    "username": None,
+                    "password": None,
+                    "client_authentication": ["client_certificate"],
+                    "rhsm_url": urljoin(host, "/rhsm"),
+                },
+                "state": "running",
+                "capabilities": capabilities,
+            }
+        }
+
+        return Response(data)
+
+
+class VersionView(APIView):
+    """
+    Returns version of the smart_proxy plugin
+    """
+
+    @extend_schema(
+        summary="Inspect version",
+        operation_id="version_read",
+    )
+    def get(self, request):
+        data = {"version": get_plugin_config("smart_proxy").version}
```

Comment:
This makes Foreman issue a warning as the version obviously doesn't match the Foreman version…

Reply:
I thought Foreman is supposed to work with "older" versions of the smart proxy...

Reply:
> This makes Foreman issue a warning as the version obviously doesn't match the Foreman version…

Yes, I've been complaining about that a few times: we need a better way to determine compatibility than version numbers.